src/device/mod.rs

```diff
@@ -4,15 +4,21 @@ use http::Uri;
 use serde::Serialize;
 use std::{collections::HashMap, str::FromStr};
 
+#[cfg(feature = "ecc608")]
 mod ecc;
 mod file;
+#[cfg(feature = "tpm")]
+mod tpm;
 
 /// A security device to work with. Security devices come in all forms. This
 /// abstracts them into one with a well defined interface for doing what this
 /// tool needs to do with them.
 #[derive(Debug)]
 pub enum Device {
+    #[cfg(feature = "ecc608")]
```

Comment:
should this not be done using he URL mechanism instead of here? I may just be missing some context here

Reply:
as in isn't ecc608 already supported? in Device?
Ah, it's the fact that ecc608 isn't optional?

Reply:
Yes. I made both tpm and ecc608 optional. Mainly to avoid installing tpm2-tss lib when it's not needed.